include/zfs_prop.h

```diff
@@ -76,10 +76,11 @@ typedef struct {
 					/* fs | vol | snap; or pool */
 	const char *pd_values;		/* string telling acceptable values */
 	const char *pd_colname;		/* column header for "zfs list" */
-	boolean_t pd_rightalign;	/* column alignment for "zfs list" */
-	boolean_t pd_visible;		/* do we list this property with the */
+	boolean_t pd_rightalign: 1;	/* column alignment for "zfs list" */
+	boolean_t pd_visible: 1;	/* do we list this property with the */
 					/* "zfs get" help message */
-	boolean_t pd_zfs_mod_supported;	/* supported by running zfs module */
+	boolean_t pd_zfs_mod_supported: 1; /* supported by running zfs module */
+	boolean_t pd_always_flex: 1;	/* never fixed-width */
```

Comment:
It feels a little out of place to make this optimization to this one structure in this PR.  If we want to tackle this I'd suggest we look in to doing something a bit more comprehensive, or perhaps targeting the structures which will have the biggest payoff.

Reply:
I mean, I'm already adding a field and breaking other ABIs so I've decided that might as well, and I'm gonna guess that across all the properties (4*sizeof(int)-1)*(95+33+41)=2535 bytes saved in .bss is, like, *definitely* worthwhile?

I was gonna update the ABI after my other libzfs-ABI-touching PR landed, but I can do it now and rebase that one later, instead, if you prefer pulling this one in first?

Reply:
Fair enough, it does all add up.